python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -106,8 +129,12 @@ def _is_redundant(index_node: Union[astroid.AssignName, astroid.Name], for_node:
     in the for loop's body.
     """
     if isinstance(index_node, astroid.AssignName):
```

Comment:
To clarify my previous comment, you should add these new test cases to the test file you've created (I didn't see any new test cases in the recent commits you made).